--- findutils-4.2.30/xargs/xargs.c.orig	2007-05-12 15:44:54.844000000 +0200
+++ findutils-4.2.30/xargs/xargs.c	2007-05-12 15:46:55.114000000 +0200
@@ -440,9 +440,9 @@
        * environment list shall not exceed {ARG_MAX}-2048 bytes.  It also 
        * specifies that it shall be at least LINE_MAX.
        */
-#if defined(ARG_MAX)
+/* #if defined(ARG_MAX)
       assert(bc_ctl.arg_max <= (ARG_MAX-2048));
-#endif
+#endif */
       assert(bc_ctl.arg_max >= LINE_MAX);
       
       bc_ctl.exec_callback = xargs_do_exec;
