--- findutils-4.2.33/xargs/xargs.c.assert	2008-02-20 22:16:25.200000000 +0100
+++ findutils-4.2.33/xargs/xargs.c	2008-02-20 22:16:48.590000000 +0100
@@ -439,9 +439,9 @@
        * environment list shall not exceed {ARG_MAX}-2048 bytes.  It also 
        * specifies that it shall be at least LINE_MAX.
        */
-#if defined(ARG_MAX)
+/* #if defined(ARG_MAX)
       assert(bc_ctl.arg_max <= (ARG_MAX-2048));
-#endif
+#endif */
 #ifdef LINE_MAX
       assert(bc_ctl.arg_max >= LINE_MAX);
 #endif
