--- findutils-4.2.27/gnulib/m4/mbchar.m4.mbchar	Sat Dec 17 18:54:03 2005
+++ findutils-4.2.27/gnulib/m4/mbchar.m4	Sat Dec 17 18:54:43 2005
@@ -9,6 +9,13 @@
 
 AC_DEFUN([gl_MBCHAR],
 [
+  AC_CHECK_HEADERS_ONCE(wchar.h wctype.h)
+
+  if test x$ac_cv_header_wchar_h = xyes \
+     && test x$ac_cv_header_wctype_h = xyes; then
+    AC_LIBOBJ(mbchar)
+  fi
+
   AC_REQUIRE([AC_GNU_SOURCE])
   :
 ])
--- findutils-4.2.27/gnulib/lib/Makefile.am.mbchar	Sat Dec 17 18:53:43 2005
+++ findutils-4.2.27/gnulib/lib/Makefile.am	Sat Dec 17 18:53:53 2005
@@ -104,7 +104,7 @@
 
 ## begin gnulib module mbchar
 
-libgnulib_a_SOURCES += mbchar.h mbchar.c
+libgnulib_a_SOURCES += mbchar.h
 
 ## end   gnulib module mbchar
 
