--- rdesktop-1.5.0/configure.ac.trio	2006-10-08 16:49:17.656800000 +0200
+++ rdesktop-1.5.0/configure.ac	2006-10-08 16:52:59.796800000 +0200
@@ -694,6 +694,9 @@
     ])
 
 
+dnl Do we need trio for snprintf
+NEED_TRIO_CHECK
+
 #
 # target-specific stuff
 #
--- rdesktop-1.5.0/rdesktop.h.trio	2006-10-08 16:59:52.036800000 +0200
+++ rdesktop-1.5.0/rdesktop.h	2006-10-08 17:00:50.446800000 +0200
@@ -31,6 +31,10 @@
 #endif
 #include <limits.h>		/* PATH_MAX */
 
+#include <trio.h>
+#define snprintf trio_snprintf
+#define vsnprintf trio_vsnprintf
+
 #define VERSION "1.5.0"
 
 #ifdef WITH_DEBUG
