--- git-1.6.5.3/git-compat-util.h.trio	2009-11-16 23:14:46.000000000 +0100
+++ git-1.6.5.3/git-compat-util.h	2009-11-28 17:45:37.858000000 +0100
@@ -465,3 +465,9 @@
 int unlink_or_warn(const char *path);
 
 #endif
+
+#ifndef HAVE_SNPRINTF
+#include <trio.h>
+#define snprintf trio_snprintf
+#define vsnprintf trio_vsnprintf
+#endif
--- git-1.6.5.3/compat/inet_ntop.c.trio	2009-11-16 23:14:46.000000000 +0100
+++ git-1.6.5.3/compat/inet_ntop.c	2009-11-28 17:47:35.158000000 +0100
@@ -15,6 +15,11 @@
  * SOFTWARE.
  */
 
+#ifndef HAVE_SNPRINTF
+#include <trio.h>
+#define snprintf trio_snprintf
+#endif
+
 #include <errno.h>
 #include <sys/types.h>
 #include <sys/socket.h>
