--- git-1.5.2.1/configure.ac.strtoumax	2007-06-05 15:23:20.616000000 +0200
+++ git-1.5.2.1/configure.ac	2007-06-05 15:24:12.196000000 +0200
@@ -271,6 +271,11 @@
 [NO_UNSETENV=],
 [NO_UNSETENV=YesPlease])
 AC_SUBST(NO_UNSETENV)
+# Define NO_STRTOUMAX if you don't have unsetenv in the C library.
+AC_CHECK_FUNC(strtoumax,
+[NO_STRTOUMAX=],
+[NO_STRTOUMAX=YesPlease])
+AC_SUBST(NO_STRTOUMAX)
 #
 # Define NO_MMAP if you want to avoid mmap.
 #
--- git-1.5.2.1/config.mak.in.strtoumax	2007-06-05 15:24:33.836000000 +0200
+++ git-1.5.2.1/config.mak.in	2007-06-05 15:25:10.876000000 +0200
@@ -39,4 +39,5 @@
 NO_SETENV=@NO_SETENV@
 NO_UNSETENV=@NO_SETENV@
 NO_ICONV=@NO_ICONV@
+NO_STRTOUMAX=@NO_STRTOUMAX@
 
