--- libpng-1.2.22/scripts/makefile.sgi.fix	2007-11-06 17:24:50.280000000 +0100
+++ libpng-1.2.22/scripts/makefile.sgi	2007-11-06 17:25:59.320000000 +0100
@@ -36,8 +36,8 @@
 #ZLIBINC=/usr/local/include
 #ZLIBLIB=/usr/local/lib
 #ZLIBINC=/usr/local/include
-ZLIBLIB=../zlib
-ZLIBINC=../zlib
+ZLIBLIB=/usr/tgcware/lib
+ZLIBINC=/usr/tgcware/include
 
 # ABI can be blank to use default for your system, -32, -o32, -n32, or -64
 # See "man abi".  zlib must be built with the same ABI.
@@ -49,10 +49,10 @@
 CFLAGS=$(ABI) -I$(ZLIBINC) -O $(WARNMORE) -DPNG_NO_MMX_CODE
 LDFLAGS_A=$(ABI) -L. -L$(ZLIBLIB) -lpng12 -lz -lm
 LDFLAGS=$(ABI) -L. -L$(ZLIBLIB) -lpng -lz -lm
-LDSHARED=cc $(ABI) -shared -soname $(LIBSOMAJ) \
+LDSHARED=cc $(ABI) -Wl,-rpath,/usr/tgcware/lib -shared -soname $(LIBSOMAJ) \
   -set_version sgi$(PNGMAJ).0
-LDLEGACY=cc $(ABI) -shared -soname $(OLDSOMAJ) \
-  -set_version sgi$3.0
+LDLEGACY=cc $(ABI) -Wl,-rpath,/usr/tgcware/lib -shared -soname $(OLDSOMAJ) \
+  -set_version sgi3.0
 # See "man dso" for info about shared objects
 
 INCPATH=$(prefix)/include
