--- libpng-1.2.12/scripts/makefile.sggcc.orig	2006-09-30 12:59:57.498800000 +0200
+++ libpng-1.2.12/scripts/makefile.sggcc	2006-09-30 13:02:03.308800000 +0200
@@ -1,4 +1,4 @@
-# makefile for libpng.a and libpng12.so, SGI IRIX with 'cc'
+# makefile for libpng.a and libpng12.so, SGI IRIX with 'gcc'
 # Copyright (C) 2001-2002, 2006 Glenn Randers-Pehrson
 # Copyright (C) 1995 Guy Eric Schalnat, Group 42, Inc.
 # For conditions of distribution and use, see copyright notice in png.h
@@ -34,22 +34,20 @@
 # Where the zlib library and include files are located
 #ZLIBLIB=/usr/local/lib32
 #ZLIBINC=/usr/local/include
-#ZLIBLIB=/usr/local/lib
-#ZLIBINC=/usr/local/include
-ZLIBLIB=../zlib
-ZLIBINC=../zlib
+ZLIBLIB=/usr/tgcware/lib
+ZLIBINC=/usr/tgcware/include
 
 # ABI can be blank to use default for your system, -32, -o32, -n32, or -64
 # See "man abi".  zlib must be built with the same ABI.
 ABI=
 
 WARNMORE= # -g -DPNG_DEBUG=5
-CFLAGS=$(ABI) -I$(ZLIBINC) -O2 $(WARNMORE) -fPIC -mabi=n32
+CFLAGS=$(ABI) -I$(ZLIBINC) -O2 $(WARNMORE) -fPIC
 LDFLAGS=$(ABI) -L. -L$(ZLIBLIB) -lpng -lz -lm
-LDSHARED=cc $(ABI) -shared -soname $(LIBSOMAJ) \
-  -set_version sgi$(PNGMAJ).0
-LDLEGACY=cc $(ABI) -shared -soname $(OLDSOMAJ) \
-  -set_version sgi$3.0
+LDSHARED=gcc $(ABI) -Wl,-rpath,/usr/tgcware/lib -shared -Wl,-soname,$(LIBSOMAJ) \
+  -Wl,-set_version,sgi$(PNGMAJ).0
+LDLEGACY=gcc $(ABI) -Wl,-rpath,/usr/tgcware/lib -shared -Wl,-soname,$(OLDSOMAJ) \
+  -Wl,-set_version,sgi3.0
 # See "man dso" for info about shared objects
 
 INCPATH=$(prefix)/include
