--- arts-1.4.0/configure.sgiaudio	Mon Mar 21 10:24:28 2005
+++ arts-1.4.0/configure	Mon Mar 21 10:24:20 2005
@@ -37734,8 +37734,8 @@
 	test "$GCC" = yes && LDFLAGS="-Wl,$LDFLAGS"
 	;;
      esac
-     echo "$as_me:$LINENO: checking for alOpenPort in -laudio" >&5
-echo $ECHO_N "checking for alOpenPort in -laudio... $ECHO_C" >&6
+     echo "$as_me:$LINENO: checking for ALopenport in -laudio" >&5
+echo $ECHO_N "checking for ALopenport in -laudio... $ECHO_C" >&6
 if test "${ac_cv_lib_audio_alOpenPort+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -37754,11 +37754,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char alOpenPort ();
+char Alopenport ();
 int
 main ()
 {
-alOpenPort ();
+ALopenport ();
   ;
   return 0;
 }
