--- libetpan-0.45/src/Makefile.in.triolib	2006-05-13 11:17:17.529403000 +0200
+++ libetpan-0.45/src/Makefile.in	2006-05-13 11:17:48.709403000 +0200
@@ -16,7 +16,7 @@
 VERSINFO    = @API_VERSION@
 CC          = @CC@
 LDFLAGS     = @LDFLAGS@ @LIBS@ @SSLLIBS@ @LIBICONV@ \
-	@DBLIB@ @GNUTLSLIB@ @SASLLIBS@
+	@DBLIB@ @GNUTLSLIB@ @SASLLIBS@ @TRIOLIB@
 CPPFLAGS    = @CPPFLAGS@
 
 SUBLIBS = data-types/libdata-types.la \
--- libetpan-0.45/configure.in.triolib	2006-05-13 11:15:45.849403000 +0200
+++ libetpan-0.45/configure.in	2006-05-13 11:16:55.469403000 +0200
@@ -62,6 +62,12 @@
 
 dnl we need snprintf
 NEED_TRIO_CHECK
+if test "x$ac_cv_need_trio" != "xno"; then
+    TRIOLIB="-ltrio"
+else
+    TRIOLIB=""
+fi
+AC_SUBST(TRIOLIB)
 
 dnl Network libs
 AC_CHECK_FUNC(connect, checksocket=no, checksocket=yes)
