--- libetpan-0.47/src/Makefile.in.cppflags	2006-10-21 17:03:29.730400000 +0200
+++ libetpan-0.47/src/Makefile.in	2006-10-21 17:03:58.370400000 +0200
@@ -16,6 +16,7 @@
 VERSINFO    = @API_VERSION@
 CC          = @CC@
 LDFLAGS     = @LDFLAGS@ @LIBS@ @SSLLIBS@ @LIBICONV@ @DBLIB@ @GNUTLSLIB@ @SASLLIBS@
+CPPFLAGS    = @CPPFLAGS@
 
 SUBLIBS = data-types/libdata-types.la \
 	low-level/liblow-level.la \
@@ -34,7 +35,7 @@
 all: $(TARGET)
 
 $(TARGET): Makefile $(SUBLIBS)
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(SUBLIBS) \
+	$(LIBTOOL) --mode=link $(CC) $(CPPFLAGS) $(LDFLAGS) -o $@ $(SUBLIBS) \
 	  -rpath $(libdir) -version-info $(VERSINFO)
 
 $(SUBLIBS): all-recursive
