--- pango-1.20.5/pango-view/viewer-render.c.sgicc	2008-11-16 19:50:02.247402000 +0100
+++ pango-1.20.5/pango-view/viewer-render.c	2008-11-16 19:54:31.857402000 +0100
@@ -579,8 +579,8 @@
   {
     {"no-auto-dir",	0, G_OPTION_FLAG_REVERSE, G_OPTION_ARG_NONE,	&opt_auto_dir,
      "No layout direction according to contents",			NULL},
-    {"backend",		0, backend_flag, G_OPTION_ARG_CALLBACK,		&parse_backend,
-     backend_desc,					     backend_options},
+    {"backend",		0, 0, G_OPTION_ARG_CALLBACK,		&parse_backend,
+     0,					     0},
     {"no-display",	'q', G_OPTION_FLAG_REVERSE, G_OPTION_ARG_NONE,	&opt_display,
      "Do not display (just write to file or whatever)",			NULL},
     {"dpi",		0, 0, G_OPTION_ARG_INT,				&opt_dpi,
@@ -640,6 +640,10 @@
   GOptionContext *context;
   size_t len;
 
+  entries[1].flags = backend_flag;
+  entries[1].description = backend_desc;
+  entries[1].arg_description = backend_options;
+
   prog_name = g_path_get_basename (argv[0]);
   context = g_option_context_new ("- FILE");
   g_option_context_add_main_entries (context, entries, NULL);
