--- pango-1.20.5/configure.nolibtoolrpath	2008-11-17 20:35:23.427402000 +0100
+++ pango-1.20.5/configure	2008-11-17 20:35:43.417402000 +0100
@@ -9651,7 +9651,7 @@
   shlibpath_overrides_runpath=no
   sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 # No shared lib support for Linux oldld, aout, or coff.
