--- Python-2.7.1/Lib/distutils/unixccompiler.py.linker	2011-05-15 11:46:20.247376000 +0200
+++ Python-2.7.1/Lib/distutils/unixccompiler.py	2011-05-15 12:03:59.234000000 +0200
@@ -290,9 +290,13 @@
             if self._is_gcc(compiler):
                 return ["-Wl,+s", "-L" + dir]
             return ["+s", "-L" + dir]
-        elif sys.platform[:7] == "irix646" or sys.platform[:6] == "osf1V5":
+        elif sys.platform[:6] == "osf1V5":
             return ["-rpath", dir]
-        elif self._is_gcc(compiler):
+        # For IRIX 5.x/6.x any combination of compilers and tools should support
+        # -Wl,-rpath while -rpath is only supported by gcc.
+        elif sys.platform[:4] == "irix":
+            return ["-Wl,-rpath", dir]
+        elif self._is_gcc(compiler):
             return "-Wl,-R" + dir
         else:
             return "-R" + dir
