--- jpeg-6b/ltconfig.orig	2003-12-21 14:50:27.670000000 +0100
+++ jpeg-6b/ltconfig	2003-12-21 14:51:25.770000000 +0100
@@ -1187,8 +1187,8 @@
 
 irix5* | irix6*)
   version_type=osf
-  soname_spec='${libname}${release}.so'
-  library_names_spec='${libname}${release}.so$versuffix $libname.so'
+  soname_spec='${libname}${release}.so.$major'
+  library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so.$major $libname.so'
   shlibpath_var=LD_LIBRARY_PATH
   ;;
 
