--- gcc-3.4.6/gcc/config/mips/t-iris5-6.orig	2007-11-14 18:46:47.000000000 +0100
+++ gcc-3.4.6/gcc/config/mips/t-iris5-6	2007-11-14 18:49:21.000000000 +0100
@@ -8,7 +8,7 @@
 SHLIB_SLIBDIR_QUAL = @shlib_slibdir_qual@
 
 SHLIB_LINK = $(GCC_FOR_TARGET) $(LIBGCC2_CFLAGS) -shared -nodefaultlibs \
-        -Wl,-soname,$(SHLIB_SONAME) \
+        -Wl,-soname,$(SHLIB_SONAME) -Wl,-no_rqs \
 	-o $(SHLIB_NAME).tmp @multilib_flags@ $(SHLIB_OBJS) -lc && \
 	rm -f $(SHLIB_SOLINK) && \
 	if [ -f $(SHLIB_NAME) ]; then \
