--- gmp-4.2.4/configure.orig	2008-09-11 22:03:04.000000000 +0200
+++ gmp-4.2.4/configure	2008-11-11 21:25:17.530333000 +0100
@@ -3870,7 +3870,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: " >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -3924,7 +3924,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: function pointer return" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -3980,7 +3980,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: cmov instruction" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4037,7 +4037,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: double -> ulong conversion" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4092,7 +4092,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: double negation" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4148,7 +4148,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: double -> float conversion" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4233,7 +4233,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: gnupro alpha ev6 char spilling" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4284,7 +4284,7 @@
 int k; int foo () { __builtin_alloca (k); }
 EOF
   echo "Test compile: __builtin_alloca availability" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4330,7 +4330,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: alloca array" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4408,7 +4408,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: abs int -> double conversion" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4472,7 +4472,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: long long reliability test 1" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4532,7 +4532,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: long long reliability test 2" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4631,7 +4631,7 @@
 
 EOF
   echo "Test compile: mpn_lshift_com optimization" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4739,7 +4739,7 @@
 
 EOF
   echo "Test compile: mpn_lshift_com optimization 2" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4798,7 +4798,7 @@
 int bar () { return foo; }
 EOF
 tmp_got_emitted=no
-gmp_compile="$cc $cflags $cppflags -fPIC -S conftest.c >&5 2>&1"
+gmp_compile="$cc $cflags $cppflags $ldflags -fPIC -S conftest.c >&5 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4872,7 +4872,7 @@
 	.byte	254, 220, 186, 152, 118, 84, 50, 16
 EOF
 tmp_got_good=yes
-gmp_compile="$cc $cflags $cppflags -fPIC -o conftest.o -c conftest.s >&5 2>&1"
+gmp_compile="$cc $cflags $cppflags $ldflags -fPIC -o conftest.o -c conftest.s >&5 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -4922,7 +4922,7 @@
 int main () { return 0; }
 EOF
 gmp_prog_cc_works=no
-gmp_compile="$cc $cflags $cppflags -c conftest.c >&5"
+gmp_compile="$cc $cflags $cppflags $ldflags -c conftest.c >&5"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5206,7 +5206,7 @@
 	xorl	%eax, %eax
 	ret
 EOF
-  gmp_compile="$cc $cflags $cppflags conftest.s -o conftest >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags conftest.s -o conftest >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5335,7 +5335,7 @@
 cat >conftest.c <<EOF
 EOF
 echo "with empty conftest.c" >&5
-gmp_compile="$cc $cflags $cppflags -c conftest.c >&5 2>&1"
+gmp_compile="$cc $cflags $cppflags $ldflags -c conftest.c >&5 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5343,7 +5343,7 @@
   (exit $ac_status); }; then : ;
 else
   # empty fails
-  gmp_compile="$cc $cflags $cppflags -Wa,-oldas -c conftest.c >&5 2>&1"
+  gmp_compile="$cc $cflags $cppflags $ldflags -Wa,-oldas -c conftest.c >&5 2>&1"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5379,7 +5379,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: " >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5433,7 +5433,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: function pointer return" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5489,7 +5489,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: cmov instruction" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5546,7 +5546,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: double -> ulong conversion" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5601,7 +5601,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: double negation" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5657,7 +5657,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: double -> float conversion" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5742,7 +5742,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: gnupro alpha ev6 char spilling" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5793,7 +5793,7 @@
 int k; int foo () { __builtin_alloca (k); }
 EOF
   echo "Test compile: __builtin_alloca availability" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5839,7 +5839,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: alloca array" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5917,7 +5917,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: abs int -> double conversion" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -5981,7 +5981,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: long long reliability test 1" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -6041,7 +6041,7 @@
 int main () { return 0; }
 EOF
   echo "Test compile: long long reliability test 2" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -6140,7 +6140,7 @@
 
 EOF
   echo "Test compile: mpn_lshift_com optimization" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -6248,7 +6248,7 @@
 
 EOF
   echo "Test compile: mpn_lshift_com optimization 2" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
+  gmp_compile="$cc $cflags $cppflags $ldflags $flag conftest.c >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -6307,7 +6307,7 @@
 int bar () { return foo; }
 EOF
 tmp_got_emitted=no
-gmp_compile="$cc $cflags $cppflags $flag -fPIC -S conftest.c >&5 2>&1"
+gmp_compile="$cc $cflags $cppflags $ldflags $flag -fPIC -S conftest.c >&5 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -6381,7 +6381,7 @@
 	.byte	254, 220, 186, 152, 118, 84, 50, 16
 EOF
 tmp_got_good=yes
-gmp_compile="$cc $cflags $cppflags $flag -fPIC -o conftest.o -c conftest.s >&5 2>&1"
+gmp_compile="$cc $cflags $cppflags $ldflags $flag -fPIC -o conftest.o -c conftest.s >&5 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -8418,7 +8418,7 @@
 int main (void) { return 0; }
 EOF
   echo "Test compile: " >&5
-  gmp_cxxcompile="$CXX $CPPFLAGS $CXXFLAGS conftest.cc >&5"
+  gmp_cxxcompile="$CXX $CPPFLAGS $CXXFLAGS $LDFLAGS conftest.cc >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_cxxcompile\"") >&5
   (eval $gmp_cxxcompile) 2>&5
   ac_status=$?
@@ -8458,7 +8458,7 @@
 int main (void) { return 0; }
 EOF
   echo "Test compile: namespace" >&5
-  gmp_cxxcompile="$CXX $CPPFLAGS $CXXFLAGS conftest.cc >&5"
+  gmp_cxxcompile="$CXX $CPPFLAGS $CXXFLAGS $LDFLAGS conftest.cc >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_cxxcompile\"") >&5
   (eval $gmp_cxxcompile) 2>&5
   ac_status=$?
@@ -8504,7 +8504,7 @@
 int main (void) { return 0; }
 EOF
   echo "Test compile: std iostream" >&5
-  gmp_cxxcompile="$CXX $CPPFLAGS $CXXFLAGS conftest.cc >&5"
+  gmp_cxxcompile="$CXX $CPPFLAGS $CXXFLAGS $LDFLAGS conftest.cc >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_cxxcompile\"") >&5
   (eval $gmp_cxxcompile) 2>&5
   ac_status=$?
@@ -26346,7 +26346,7 @@
   cat >conftest.c <<EOF
 void *foo (int x) __attribute__ ((malloc));
 EOF
-gmp_compile="$CC $CFLAGS $CPPFLAGS -c conftest.c >conftest.out 2>&1"
+gmp_compile="$CC $CFLAGS $CPPFLAGS $LDFLAGS -c conftest.c >conftest.out 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -27383,7 +27383,7 @@
   { '\376', '\334', '\272', '\230', '\166', '\124', '\062', '\020' },
 };
 EOF
-gmp_compile="$CC $CFLAGS $CPPFLAGS -c conftest.c >&5 2>&1"
+gmp_compile="$CC $CFLAGS $CPPFLAGS $LDFLAGS -c conftest.c >&5 2>&1"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -29133,7 +29133,7 @@
 cat >conftest.c <<EOF
 int gurkmacka;
 EOF
-gmp_compile="$CC $CFLAGS $CPPFLAGS -c conftest.c >&5"
+gmp_compile="$CC $CFLAGS $CPPFLAGS $LDFLAGS -c conftest.c >&5"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -29213,7 +29213,7 @@
 EOF
 echo "Test program:" >&5
 cat conftest.c >&5
-gmp_compile="$CC $CFLAGS $CPPFLAGS -S conftest.c >&5"
+gmp_compile="$CC $CFLAGS $CPPFLAGS $LDFLAGS -S conftest.c >&5"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -29876,7 +29876,7 @@
 EOF
 echo "Test program:" >&5
 cat conftest.c >&5
-gmp_compile="$CC $CFLAGS $CPPFLAGS -S conftest.c >&5"
+gmp_compile="$CC $CFLAGS $CPPFLAGS $LDFLAGS -S conftest.c >&5"
 if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
@@ -30160,7 +30160,7 @@
 ${tmp_gsym_prefix}main$gmp_cv_asm_label_suffix
 	addl	$ ${tmp_underscore}_GLOBAL_OFFSET_TABLE_, %ebx
 EOF
-  gmp_compile="$CCAS $CFLAGS $CPPFLAGS $lt_prog_compiler_pic conftest.s >&5 && $CC $CFLAGS $CPPFLAGS $lt_prog_compiler_pic conftest.$OBJEXT >&5"
+  gmp_compile="$CCAS $CFLAGS $CPPFLAGS $LDFLAGS $lt_prog_compiler_pic conftest.s >&5 && $CC $CFLAGS $CPPFLAGS $LDFLAGS $lt_prog_compiler_pic conftest.$OBJEXT >&5"
   if { (eval echo "$as_me:$LINENO: \"$gmp_compile\"") >&5
   (eval $gmp_compile) 2>&5
   ac_status=$?
