
Fix for gcc 3.x:

--- gmp-4.1.2/tests/cxx/t-locale.cc.cxxtests
+++ gmp-4.1.2/tests/cxx/t-locale.cc
@@ -21,7 +21,7 @@
 
 #include "config.h"
 
-#include <strstream.h>
+#include <strstream>
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -91,7 +91,7 @@
               sprintf (str+strlen(str), data[j].str, decimal_point);
 
               mpf_set_d (f, 123.0);
-              istrstream i (str);
+              std::istrstream i (str);
               if (! (i >> f))
                 {
                   printf ("operator>> input error\n");
@@ -128,7 +128,7 @@
     mpf_init (f);
     decimal_point = ",";
     mpf_set_d (f, 1.5);
-    ostrstream  got;
+    std::ostrstream  got;
     got << f << '\0';
     mpf_clear (f);
     if (strcmp (got.str(), "1,5") != 0)
