--- freeglut-2.2.0/src/freeglut_font.c.orig	Wed Mar 23 10:10:54 2005
+++ freeglut-2.2.0/src/freeglut_font.c	Wed Mar 23 10:11:35 2005
@@ -114,7 +114,7 @@
      */
     face = font->Characters[ character - 1 ];
 
-    glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT );
+    /* glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT ); */
     glPixelStorei( GL_UNPACK_SWAP_BYTES,  GL_FALSE );
     glPixelStorei( GL_UNPACK_LSB_FIRST,   GL_FALSE );
     glPixelStorei( GL_UNPACK_ROW_LENGTH,  0        );
@@ -127,7 +127,7 @@
         ( float )( face[ 0 ] ), 0.0,  /* The raster advance -- inc. x,y */
         ( face + 1 )                  /* The packed bitmap data...      */
     );
-    glPopClientAttrib( );
+    /* glPopClientAttrib( ); */
 }
 
 void FGAPIENTRY glutBitmapString( void* fontID, const unsigned char *string )
@@ -138,7 +138,7 @@
     float raster_position[ 4 ];
 
     glGetFloatv ( GL_CURRENT_RASTER_POSITION, raster_position );
-    glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT );
+    /* glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT ); */
     glPixelStorei( GL_UNPACK_SWAP_BYTES,  GL_FALSE );
     glPixelStorei( GL_UNPACK_LSB_FIRST,   GL_FALSE );
     glPixelStorei( GL_UNPACK_ROW_LENGTH,  0        );
@@ -168,7 +168,7 @@
                 ( face + 1 )                 /* The packed bitmap data...    */
             );
         }
-    glPopClientAttrib( );
+    /* glPopClientAttrib( ); */
 }
 
 /*
