--- freeglut-2.4.0/src/freeglut_font.c.orig	2005-10-22 19:34:04.212463947 +0200
+++ freeglut-2.4.0/src/freeglut_font.c	2005-10-22 19:36:02.524186210 +0200
@@ -112,7 +112,7 @@
      */
     face = font->Characters[ character ];
 
-    glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT );
+    /* glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT ); */
     glPixelStorei( GL_UNPACK_SWAP_BYTES,  GL_FALSE );
     glPixelStorei( GL_UNPACK_LSB_FIRST,   GL_FALSE );
     glPixelStorei( GL_UNPACK_ROW_LENGTH,  0        );
@@ -125,7 +125,7 @@
         ( float )( face[ 0 ] ), 0.0,  /* The raster advance -- inc. x,y */
         ( face + 1 )                  /* The packed bitmap data...      */
     );
-    glPopClientAttrib( );
+    /* glPopClientAttrib( ); */
 }
 
 void FGAPIENTRY glutBitmapString( void* fontID, const unsigned char *string )
@@ -139,7 +139,7 @@
     if ( !string || ! *string )
         return;
 
-    glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT );
+    /* glPushClientAttrib( GL_CLIENT_PIXEL_STORE_BIT ); */
     glPixelStorei( GL_UNPACK_SWAP_BYTES,  GL_FALSE );
     glPixelStorei( GL_UNPACK_LSB_FIRST,   GL_FALSE );
     glPixelStorei( GL_UNPACK_ROW_LENGTH,  0        );
@@ -172,7 +172,7 @@
             x += ( float )( face[ 0 ] );
         }
 
-    glPopClientAttrib( );
+    /* glPopClientAttrib( ); */
 }
 
 /*
