--- kdelibs-3.4.0/acinclude.m4.socklen_t	Sat Mar 26 13:31:48 2005
+++ kdelibs-3.4.0/acinclude.m4	Sat Mar 26 13:34:22 2005
@@ -3376,6 +3376,8 @@
    fi
    AC_DEFINE_UNQUOTED(kde_socklen_t, $kde_cv_socklen_t_equiv,
                      [type to use in place of socklen_t if not defined])
+   AC_DEFINE_UNQUOTED(socklen_t, $kde_cv_socklen_t_equiv,
+                     [type to use in place of socklen_t if not defined])
    AC_DEFINE_UNQUOTED(ksize_t, $kde_cv_socklen_t_equiv,
                      [type to use in place of socklen_t if not defined (deprecated, use kde_socklen_t)])
 ])
