--- SDL-1.2.11/src/thread/pthread/SDL_sysmutex.c.ifdef	2006-07-01 20:57:33.852378000 +0200
+++ SDL-1.2.11/src/thread/pthread/SDL_sysmutex.c	2006-07-01 20:58:00.996948000 +0200
@@ -32,7 +32,7 @@
 
 struct SDL_mutex {
 	pthread_mutex_t id;
-#if FAKE_RECURSIVE_MUTEX
+#ifdef FAKE_RECURSIVE_MUTEX
 	int recursive;
 	pthread_t owner;
 #endif
@@ -77,7 +77,7 @@
 int SDL_mutexP(SDL_mutex *mutex)
 {
 	int retval;
-#if FAKE_RECURSIVE_MUTEX
+#ifdef FAKE_RECURSIVE_MUTEX
 	pthread_t this_thread;
 #endif
 
@@ -87,7 +87,7 @@
 	}
 
 	retval = 0;
-#if FAKE_RECURSIVE_MUTEX
+#ifdef FAKE_RECURSIVE_MUTEX
 	this_thread = pthread_self();
 	if ( mutex->owner == this_thread ) {
 		++mutex->recursive;
@@ -123,7 +123,7 @@
 	}
 
 	retval = 0;
-#if FAKE_RECURSIVE_MUTEX
+#ifdef FAKE_RECURSIVE_MUTEX
 	/* We can only unlock the mutex if we own it */
 	if ( pthread_self() == mutex->owner ) {
 		if ( mutex->recursive ) {
