--- perl-5.8.7/hints/irix_6.sh.orig	2005-10-23 11:01:39.411258889 +0200
+++ perl-5.8.7/hints/irix_6.sh	2005-10-23 11:03:54.570378439 +0200
@@ -189,20 +189,6 @@
 	# It seems gcc can build Irix shlibs, but of course it needs
 	# -shared.  Otherwise you get link errors looking for main().
 	lddlflags="$lddlflags -shared"
-	case "`uname -s`" in
-	# Without the -mabi=64 gcc in 64-bit IRIX has problems passing
-	# and returning small structures.  This affects inet_*() and semctl().
-	# See http://reality.sgi.com/ariel/freeware/gcc-2.8.1-notes.html
-	# for more information.  Reported by Lionel Cons <lionel.cons@cern.ch>.
-	IRIX64)	ccflags="$ccflags -mabi=64"
-		ldflags="$ldflags -mabi=64 -L/usr/lib64"
-		lddlflags="$lddlflags -mabi=64"
-		;;
-	*)	ccflags="$ccflags -DIRIX32_SEMUN_BROKEN_BY_GCC"
-                # XXX Note: It is possible that turning off usemallocwrap is
-                # needed here; insufficient data! - Allen
-		;;
-	esac
 	;;
 *)
 	# this is needed to force the old-32 paths
