--- perl-5.8.5/hints/irix_6.sh.gcc34	Sat Nov 13 00:27:10 2004
+++ perl-5.8.5/hints/irix_6.sh	Sat Nov 13 13:04:11 2004
@@ -178,18 +178,6 @@
 	# It seems gcc can build Irix shlibs, but of course it needs
 	# -shared.  Otherwise you get link errors looking for main().
 	lddlflags="$lddlflags -shared"
-	case "`uname -s`" in
-	# Without the -mabi=64 gcc in 64-bit IRIX has problems passing
-	# and returning small structures.  This affects inet_*() and semctl().
-	# See http://reality.sgi.com/ariel/freeware/gcc-2.8.1-notes.html
-	# for more information.  Reported by Lionel Cons <lionel.cons@cern.ch>.
-	IRIX64)	ccflags="$ccflags -mabi=64"
-		ldflags="$ldflags -mabi=64 -L/usr/lib64"
-		lddlflags="$lddlflags -mabi=64"
-		;;
-	*)	ccflags="$ccflags -DIRIX32_SEMUN_BROKEN_BY_GCC"
-		;;
-	esac
 	;;
 *)
 	# this is needed to force the old-32 paths
 	#  since the system default can be changed.
