--- gimp-2.2.10/aclocal.m4.norpath	2006-04-12 20:22:29.516489000 +0200
+++ gimp-2.2.10/aclocal.m4	2006-04-12 20:30:18.906489000 +0200
@@ -2473,7 +2473,7 @@
   shlibpath_overrides_runpath=no
   sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 # No shared lib support for Linux oldld, aout, or coff.
