--- m4-1.4.15/tests/test-langinfo.c.orig	2010-03-08 22:47:54.000000000 +0100
+++ m4-1.4.15/tests/test-langinfo.c	2010-09-10 20:31:25.650000000 +0200
@@ -32,7 +32,9 @@
     D_T_FMT,
     D_FMT,
     T_FMT,
+#ifdef T_FMT_AMPM
     T_FMT_AMPM,
+#endif
     AM_STR,
     PM_STR,
     DAY_1,
@@ -78,11 +80,13 @@
     ERA_D_T_FMT,
     ERA_T_FMT,
     ALT_DIGITS,
-    /* nl_langinfo items of the LC_MONETARY category */
-    CRNCYSTR,
     /* nl_langinfo items of the LC_MESSAGES category */
+#ifdef YESEXPR
     YESEXPR,
-    NOEXPR
+    NOEXPR,
+#endif
+    /* nl_langinfo items of the LC_MONETARY category */
+    CRNCYSTR
   };
 
 int
--- m4-1.4.15/tests/test-nl_langinfo.c.orig	2010-03-08 22:47:54.000000000 +0100
+++ m4-1.4.15/tests/test-nl_langinfo.c	2010-09-10 20:33:24.230000000 +0200
@@ -56,7 +56,9 @@
   ASSERT (strlen (nl_langinfo (D_T_FMT)) > 0);
   ASSERT (strlen (nl_langinfo (D_FMT)) > 0);
   ASSERT (strlen (nl_langinfo (T_FMT)) > 0);
+#ifdef T_FMT_AMPM
   ASSERT (strlen (nl_langinfo (T_FMT_AMPM)) >= (pass == 0 ? 1 : 0));
+#endif
   ASSERT (strlen (nl_langinfo (AM_STR)) >= (pass == 0 ? 1 : 0));
   ASSERT (strlen (nl_langinfo (PM_STR)) >= (pass == 0 ? 1 : 0));
   ASSERT (strlen (nl_langinfo (DAY_1)) > 0);
@@ -108,8 +110,10 @@
     ASSERT (strlen (currency) >= (pass > 0 ? 1 : 0));
   }
   /* nl_langinfo items of the LC_MESSAGES category */
+#ifdef YESEXPR
   ASSERT (strlen (nl_langinfo (YESEXPR)) > 0);
   ASSERT (strlen (nl_langinfo (NOEXPR)) > 0);
+#endif
 
   return 0;
 }
