--- openssh-4.3p2/openbsd-compat/getrrsetbyname.c	12 Nov 2005 03:06:29 -0000	1.17
+++ openssh-4.3p2/openbsd-compat/getrrsetbyname.c	2 May 2006 00:51:49 -0000
@@ -60,6 +60,10 @@ extern int h_errno;
 # undef _THREAD_PRIVATE
 #endif
 #define _THREAD_PRIVATE(a,b,c) (c)
+
+/* to avoid conflicts where a platform already has _res */
+#define _res	_compat_res
+
 struct __res_state _res;
 
 /* Necessary functions and macros */
