--- openssh-5.2p1/channels.c.no_in_port_t	2009-03-16 19:27:42.110000000 +0100
+++ openssh-5.2p1/channels.c	2009-03-16 19:29:04.790000000 +0100
@@ -2469,7 +2469,7 @@
 	struct addrinfo hints, *ai, *aitop;
 	const char *host, *addr;
 	char ntop[NI_MAXHOST], strport[NI_MAXSERV];
-	in_port_t *lport_p;
+	u_int16_t *lport_p;
 
 	host = (type == SSH_CHANNEL_RPORT_LISTENER) ?
 	    listen_addr : host_to_connect;
