--- openssh/defines.h.orig	2011-08-14 11:47:19.098373000 +0200
+++ openssh/defines.h	2011-08-14 11:47:34.818373000 +0200
@@ -45,8 +45,8 @@
 /*
  * Definitions for IP type of service (ip_tos)
  */
-#include <netinet/in_systm.h>
-#include <netinet/ip.h>
+/* #include <netinet/in_systm.h>
+#include <netinet/ip.h> */
 #ifndef IPTOS_LOWDELAY
 # define IPTOS_LOWDELAY          0x10
 # define IPTOS_THROUGHPUT        0x08
