--- kdebase-3.5.7/ksysguard/ksysguardd/Irix/cpu.c.increorder	2007-06-02 10:56:01.622000000 +0200
+++ kdebase-3.5.7/ksysguard/ksysguardd/Irix/cpu.c	2007-06-02 10:56:08.652000000 +0200
@@ -28,10 +28,11 @@
 #include <sys/sysmp.h>
 #include <sys/sysinfo.h>
 
-#include "cpu.h"
 #include "Command.h"
 #include "ksysguardd.h"
 
+#include "cpu.h"
+
 #define CPUSTATES 6
 
 long percentages(int cnt, int *out, long *new, long *old, long *diffs);
