--- kdebase-3.5.7/kdm/kfrontend/Makefile.in.x11	2007-06-02 17:52:45.882000000 +0200
+++ kdebase-3.5.7/kdm/kfrontend/Makefile.in	2007-06-02 18:00:29.242000000 +0200
@@ -663,7 +663,7 @@
     kgapp.cpp
 
 kdm_greet_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kdm_greet_LDADD = themer/libkdmthemer.a $(LIB_KDEUI) $(XTESTLIB) $(LIBPOSIX4)
+kdm_greet_LDADD = themer/libkdmthemer.a $(LIB_KDEUI) $(XTESTLIB) $(LIB_X11) $(LIB_XEXT) $(LIBPOSIX4)
 krootimage_SOURCES = krootimage.cpp
 krootimage_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 krootimage_LDADD = $(top_builddir)/kcontrol/background/libbgnd.la $(LIB_KIO)
