--- kdebase-3.5.7/kdm/kfrontend/Makefile.in.kdefakes	2007-06-02 18:06:26.582000000 +0200
+++ kdebase-3.5.7/kdm/kfrontend/Makefile.in	2007-06-02 18:06:47.482000000 +0200
@@ -672,7 +672,7 @@
 genkdmconf_LDFLAGS = $(X_LDFLAGS) $(X_RPATH)
 genkdmconf_LDADD = $(LIB_X11)
 kdmctl_SOURCES = kdmctl.c
-kdmctl_LDADD = $(LIBSOCKET)
+kdmctl_LDADD = -lkdefakes $(LIBSOCKET)
 noinst_HEADERS = \
     kdm_greet.h \
     kdmconfig.h \
