--- gtk+-2.10.7/gtk/gtkxembed.c.snprintf	2007-01-08 18:02:09.720000000 +0100
+++ gtk+-2.10.7/gtk/gtkxembed.c	2007-01-08 18:02:31.730000000 +0100
@@ -238,7 +238,7 @@
       CASE (GTK_UNGRAB_KEY);
 #undef CASE
     default:
-      snprintf (unk, 24, "UNKNOWN(%d)", message);
+      g_snprintf (unk, 24, "UNKNOWN(%d)", message);
       return unk;
     }
 }
--- gtk+-2.10.7/gtk/gtktreeview.c.snprintf	2007-01-08 18:02:41.580000000 +0100
+++ gtk+-2.10.7/gtk/gtktreeview.c	2007-01-08 18:03:09.280000000 +0100
@@ -4391,13 +4391,13 @@
 	       * less messy way).
 	       */
 	      if (is_first && is_last)
-		snprintf (new_detail, 127, "%s", detail);
+		g_snprintf (new_detail, 127, "%s", detail);
 	      else if (is_first)
-		snprintf (new_detail, 127, "%s_start", detail);
+		g_snprintf (new_detail, 127, "%s_start", detail);
 	      else if (is_last)
-		snprintf (new_detail, 127, "%s_end", detail);
+		g_snprintf (new_detail, 127, "%s_end", detail);
 	      else
-		snprintf (new_detail, 128, "%s_middle", detail);
+		g_snprintf (new_detail, 128, "%s_middle", detail);
 
 	      gtk_paint_flat_box (widget->style,
 				  event->window,
