--- xsnow-1.42/Imakefile.orig	2003-12-23 21:42:40.260000000 +0100
+++ xsnow-1.42/Imakefile	2003-12-23 21:17:47.710000000 +0100
@@ -1,7 +1,7 @@
 SYS_LIBRARIES = -lm -lXpm
 LOCAL_LIBRARIES = $(XLIB)
 DEPLIBS = $(DEPXLIB)
-SRCS = xsnow.c toon_root.c xsnow.h vroot.h
-OBJS = xsnow.o toon_root.o
+SRCS = xsnow.c toon_root.c xsnow.h vroot.h bsd-snprintf.c
+OBJS = xsnow.o toon_root.o bsd-snprintf.o
 
 ComplexProgramTarget(xsnow)
