diff -urN prngd-0.9.29/Makefile prngd-0.9.29.irix53/Makefile
--- prngd-0.9.29/Makefile	2004-07-12 17:13:35.000000000 +0200
+++ prngd-0.9.29.irix53/Makefile	2004-07-23 10:49:17.980000000 +0200
@@ -97,6 +97,12 @@
 # CFLAGS=-O3 -Wall -DNEXTSTEP -D_POSIX_SOURCE
 # LDFLAGS=-L../ossh/openbsd-compat/ -lopenbsd-compat
 
+# IRIX 5.3 w/gcc ("Tom G. Christensen" <tgc@jupiterrise.com>)
+# CFLAGS=-O3 -Wall -DIRIX53
+
+# IRIX 5.3 w/cc ("Tom G. Christensen" <tgc@jupiterrise.com>)
+# CFLAGS=-O2 -fullwarn -DIRIX53
+
 # IRIX 6.2 w/gcc ("Michael Weiser" <michael@weiser.saale-net.de>)
 # CFLAGS=-O3 -Wall -DIRIX62
 
diff -urN prngd-0.9.29/config.h prngd-0.9.29.irix53/config.h
--- prngd-0.9.29/config.h	2004-07-12 17:13:35.000000000 +0200
+++ prngd-0.9.29.irix53/config.h	2004-07-23 10:49:18.030000000 +0200
@@ -204,6 +204,20 @@
 #define PATH_SYSLOG     "/private/adm/messages"
 #endif
 
+#ifdef IRIX53
+#define HAVE_GETRUSAGE
+#define NEED_SYS_SELECT_H
+#define NEED_BSTRING_H
+#define O_RSYNC 0
+#define SOCKLEN_T       int
+#define PATH_TMP        "/tmp"
+#define PATH_VAR_TMP    "/var/tmp"
+#define PATH_PASSWD     "/etc/passwd"
+#define PATH_WTMP       "/var/adm/wtmpx"
+#define PATH_UTMP       "/var/adm/utmpx"
+#define PATH_SYSLOG     "/var/adm/SYSLOG"
+#endif
+
 #ifdef IRIX62
 #define HAVE_GETRUSAGE
 #define SOCKLEN_T       int
diff -urN prngd-0.9.29/contrib/IRIX-53/prngd.conf.irix-53 prngd-0.9.29.irix53/contrib/IRIX-53/prngd.conf.irix-53
--- prngd-0.9.29/contrib/IRIX-53/prngd.conf.irix-53	1970-01-01 01:00:00.000000000 +0100
+++ prngd-0.9.29.irix53/contrib/IRIX-53/prngd.conf.irix-53	2004-07-23 10:51:57.170000000 +0200
@@ -0,0 +1,39 @@
+# entropy gathering commands
+# Irix 5.3 & 6.2, contributed by Tom G. Christensen <tgc@jupiterrise.com>
+
+# Format is: "program-name args" path rate
+
+# The "rate" represents the number of bits of usuable entropy per 
+# byte of command output. Be conservative.
+
+"ls -alni /var/adm" /usr/bin/ls 0.02
+"ls -alni /var/mail" /usr/bin/ls 0.02
+"ls -alni /var/spool/mqueue" /usr/bin/ls 0.02
+"ls -alni /proc" /usr/bin/ls 0.02
+"ls -alni /tmp" /usr/bin/ls 0.02
+
+"netstat -an" /usr/etc/netstat 0.05
+"netstat -in" /usr/etc/netstat 0.05
+"netstat -rn" /usr/etc/netstat 0.02
+"netstat -an" /usr/etc/netstat 0.05
+"netstat -s" /usr/etc/netstat 0.02
+"netstat -is" /usr/etc/netstat 0.02
+"netstat -m" /usr/etc/netstat 0.02
+
+"ps -al" /usr/bin/ps 0.03
+"ps -efl" /usr/bin/ps 0.03
+
+"w" /usr/bsd/w 0.05
+
+"who" /usr/bin/who 0.01
+
+"last" /usr/bsd/last 0.01
+
+"df" /usr/bin/df 0.01
+"df -i" /usr/bin/df 0.01
+
+"uptime" /usr/bsd/uptime 0.01
+
+"ipcs -a" /usr/bin/ipcs 0.01
+
+"tail -200 /var/adm/SYSLOG" /usr/bin/tail 0.01
diff -urN prngd-0.9.29/prngd.h prngd-0.9.29.irix53/prngd.h
--- prngd-0.9.29/prngd.h	2004-05-17 10:24:46.000000000 +0200
+++ prngd-0.9.29.irix53/prngd.h	2004-07-23 10:49:18.100000000 +0200
@@ -23,6 +23,10 @@
 int vsnprintf(char *str, size_t count, const char *fmt, va_list args);
 #endif /* !HAVE_SNPRINTF */
 
+#ifdef NEED_BSTRING_H
+#include <bstring.h>
+#endif /* NEED_BSTRING_H */
+
 typedef struct {
   double rate;
   unsigned int badness;
