--- gawk-3.1.3/regcomp.c.lr	2003-09-22 09:38:06.000000000 +0200
+++ gawk-3.1.3/regcomp.c	2003-09-22 09:50:38.000000000 +0200
@@ -2965,7 +2965,10 @@
 #endif /* RE_ENABLE_I18N */
 				      start_elem.opr.name, syntax);
               if (BE (ret != REG_NOERROR, 0))
-               goto parse_bracket_exp_espace;
+	      {
+		*err = ret;
+		goto parse_bracket_exp_free_return;
+	      }
 	      break;
 	    default:
 	      assert (0);
