--- flac-1.1.2/configure.in.xmms	2005-11-05 21:02:03.992657044 +0100
+++ flac-1.1.2/configure.in	2005-11-05 21:03:10.712349671 +0100
@@ -117,14 +117,14 @@
 AC_DEFINE(FLAC__USE_ALTIVEC)
 fi
 
-AC_ARG_ENABLE(local-xmms-plugin,
-[  --enable-local-xmms-plugin     Install XMMS plugin to ~/.xmms/Plugins instead of system location],
-[case "${enableval}" in
-	yes) install_xmms_plugin_locally=true ;;
-	no)  install_xmms_plugin_locally=false ;;
-	*) AC_MSG_ERROR(bad value ${enableval} for --enable-local-xmms-plugin) ;;
-esac],[install_xmms_plugin_locally=false])
-AM_CONDITIONAL(FLaC__INSTALL_XMMS_PLUGIN_LOCALLY, test x$install_xmms_plugin_locally = xtrue)
+dnl AC_ARG_ENABLE(local-xmms-plugin,
+dnl [  --enable-local-xmms-plugin     Install XMMS plugin to ~/.xmms/Plugins instead of system location],
+dnl [case "${enableval}" in
+dnl 	yes) install_xmms_plugin_locally=true ;;
+dnl 	no)  install_xmms_plugin_locally=false ;;
+dnl 	*) AC_MSG_ERROR(bad value ${enableval} for --enable-local-xmms-plugin) ;;
+dnl esac],[install_xmms_plugin_locally=false])
+dnl AM_CONDITIONAL(FLaC__INSTALL_XMMS_PLUGIN_LOCALLY, test x$install_xmms_plugin_locally = xtrue)
 
 AC_ARG_ENABLE(exhaustive-tests,
 [  --enable-exhaustive-tests      Enable exhaustive testing],
@@ -157,8 +157,8 @@
 AC_DEFINE(FLAC__HAS_OGG)
 fi
 
-AM_PATH_XMMS(0.9.5.1, , AC_MSG_WARN([*** XMMS >= 0.9.5.1 not installed - xmms support will not be built]))
-AM_CONDITIONAL(FLaC__HAS_XMMS, test x$XMMS_INPUT_PLUGIN_DIR != x)
+dnl AM_PATH_XMMS(0.9.5.1, , AC_MSG_WARN([*** XMMS >= 0.9.5.1 not installed - xmms support will not be built]))
+dnl AM_CONDITIONAL(FLaC__HAS_XMMS, test x$XMMS_INPUT_PLUGIN_DIR != x)
 
 dnl check for i18n(internationalization); these are from libiconv/gettext
 AM_ICONV
