--- cairo-1.0.2/pixman/src/pixman.h.nostdint	2005-11-26 17:54:21.500756882 +0100
+++ cairo-1.0.2/pixman/src/pixman.h	2005-11-26 17:56:49.167944105 +0100
@@ -83,7 +83,7 @@
 
 #if defined (__SVR4) && defined (__sun)
 # include <sys/int_types.h>
-#elif defined (__OpenBSD__) || defined (_AIX)
+#elif defined (__OpenBSD__) || defined (_AIX) || defined (__sgi__)
 # include <inttypes.h>
 #elif defined (_MSC_VER)
   typedef __int8 int8_t;
