--- openttd-0.4.5/Makefile.icondir	2006-02-18 17:45:40.504871820 +0100
+++ openttd-0.4.5/Makefile	2006-02-18 17:46:16.039382186 +0100
@@ -911,7 +911,8 @@
 	install -d $(DATA_DIR_INSTALL)/lang \
 	           $(DATA_DIR_INSTALL)/data \
 	           $(DATA_DIR_INSTALL)/gm \
-	           $(BINARY_DIR_INSTALL)
+	           $(BINARY_DIR_INSTALL) \
+	           $(ICON_DIR_INSTALL)
 ifndef USE_HOMEDIR
 	mkdir -p $(PERSONAL_DIR)/scenario
 else
