--- libxml2-2.6.26/threads.c.irix62	2006-09-24 14:17:27.741600000 +0200
+++ libxml2-2.6.26/threads.c	2006-09-24 14:43:56.325600000 +0200
@@ -48,6 +48,17 @@
 
 #ifdef HAVE_PTHREAD_H
 
+/* For Irix 6.2 which has POSIX pthreads but defines pthread_equal as a macro
+ * which xmlInitThreads is not too happy with
+ */
+#undef pthread_equal
+int pthread_equal(pthread_t thread1, pthread_t thread2)
+{
+    if (thread1 == thread2)
+	return 1;
+    return 0;
+}
+
 static int libxml_is_threaded = -1;
 #ifdef __GNUC__
 #ifdef linux
