--- libungif-4.1.0b1/util/gif2iris.c.orig	Thu Dec 25 14:46:02 2003
+++ libungif-4.1.0b1/util/gif2iris.c	Thu Dec 25 14:46:19 2003
@@ -20,7 +20,6 @@
 
 #ifdef __MSDOS__
 #include <graphics.h>
-#include <stdlib.h>
 #include <alloc.h>
 #include <io.h>
 #include <dos.h>
@@ -30,6 +29,7 @@
 #include "gl.h"
 #include "device.h"
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <ctype.h>
 #include <string.h>
@@ -76,7 +76,7 @@
     IrisPosY = 0,
     InterlacedOffset[] = { 0, 4, 2, 1 }, /* The way Interlaced image should. */
     InterlacedJumps[] = { 8, 8, 4, 2 };    /* be read - offsets and jumps... */
-static GifColorType
+static ColorMapObject
     *ColorMap;
 
 static void Screen2Iris(GifRowType *ScreenBuffer,
@@ -227,8 +227,7 @@
     BackGround = GifFile->SBackGroundColor;
     ColorMap = (GifFile->Image.ColorMap ? GifFile->Image.ColorMap :
 				       GifFile->SColorMap);
-    ColorMapSize = 1 << (GifFile->Image.ColorMap ? GifFile->Image.BitsPerPixel :
-				                GifFile->SBitsPerPixel);
+    ColorMapSize = ColorMap->ColorCount;
     GifQprintf("\n");
     Screen2Iris(ScreenBuffer, GifFile->SWidth, GifFile->SHeight);
 
@@ -249,7 +248,8 @@
     short Val;
     int i, j;
     unsigned long *IrisScreenBuffer, *PBuffer;
-
+    GifColorType *ColorMapEntry = ColorMap->Colors;
+    
     if (ScreenWidth > XMAXSCREEN + 1 ||
 	ScreenHeight > YMAXSCREEN + 1)
 	GIF_EXIT("Input image is too big.");
@@ -272,9 +272,9 @@
     PBuffer = IrisScreenBuffer;
     for (i = ScreenHeight - 1; i >= 0; i--)
 	for (j = 0; j < ScreenWidth; j++)
-	    *PBuffer++ = ColorMap[ScreenBuffer[i][j]].Red +
-		         (ColorMap[ScreenBuffer[i][j]].Green << 8) +
-		         (ColorMap[ScreenBuffer[i][j]].Blue << 16);
+	    *PBuffer++ = ColorMapEntry[ScreenBuffer[i][j]].Red +
+		         (ColorMapEntry[ScreenBuffer[i][j]].Green << 8) +
+		         (ColorMapEntry[ScreenBuffer[i][j]].Blue << 16);
 
     reshapeviewport();
     lrectwrite(0, 0, ScreenWidth - 1, ScreenHeight - 1, IrisScreenBuffer);
