--- libungif-4.1.0b1/lib/dev2gif.c.orig	Thu Dec 25 12:01:14 2003
+++ libungif-4.1.0b1/lib/dev2gif.c	Thu Dec 25 12:01:38 2003
@@ -25,13 +25,13 @@
 #include <graphics.h>
 #endif /* __MSDOS__ */
 
-#ifdef HAVE_LIBGL_S
-#include <gl/gl.h>
-#endif /* HAVE_LIBGL_S */
-
 #ifdef HAVE_LIBX11
 #include <X11/Xlib.h>
 #endif /* HAVE_LIBX11 */
+
+#ifdef HAVE_LIBGL_S
+#include <gl/gl.h>
+#endif /* HAVE_LIBGL_S */
 
 #include <stdlib.h>
 #include <stdio.h>
