--- gnutls-1.2.10/src/tests.c.siginterrupt	Sun May  7 10:57:56 2006
+++ gnutls-1.2.10/src/tests.c	Sun May  7 11:00:19 2006
@@ -54,6 +54,24 @@
 static int sfree = 0;
 static int handshake_output = 0;
 
+/* Irix 5.3 lacks siginterrupt */
+/* This implementation was lifted from bash 3.1 */
+static int
+siginterrupt (sig, flag)
+     int sig, flag;
+{
+  struct sigaction act;
+
+  sigaction (sig, (struct sigaction *)NULL, &act);
+
+  if (flag)
+    act.sa_flags &= ~SA_RESTART;
+  else
+    act.sa_flags |= SA_RESTART;
+
+  return (sigaction (sig, &act, (struct sigaction *)NULL));
+}
+
 int do_handshake(gnutls_session session)
 {
     int ret, alert;
