--- libao-1.1.0/configure.ac.orig	2011-12-17 15:17:07.921600000 +0100
+++ libao-1.1.0/configure.ac	2011-12-17 15:20:32.765600000 +0100
@@ -145,6 +145,13 @@
 dnl ==============================
 
 dnl ==============================
+dnl Checks for functions
+dnl ==============================
+
+dnl We need snprintf
+NEED_TRIO_CHECK
+
+dnl ==============================
 dnl Select proper plugin options
 dnl ==============================
 case $host in
