--- curl-7.24.0/tests/server/sws.c.orig	2012-01-23 16:31:30.000000000 +0100
+++ curl-7.24.0/tests/server/sws.c	2012-03-10 22:34:18.815600000 +0100
@@ -65,6 +65,16 @@
 /* include memdebug.h last */
 #include "memdebug.h"
 
+#ifndef SHUT_RDWR
+#define SHUT_RDWR 2
+#endif
+#ifndef SHUT_WR
+#define SHUT_WR 1
+#endif
+#ifndef SHUT_RD
+#define SHUT_RD 0
+#endif
+
 #if !defined(CURL_SWS_FORK_ENABLED) && defined(HAVE_FORK)
 /*
  * The normal sws build for the plain standard curl test suite has no use for
