--- sylpheed-claws-2.2.0/configure.ac.morechecks	2006-05-13 15:01:05.859403000 +0200
+++ sylpheed-claws-2.2.0/configure.ac	2006-05-13 15:05:25.189403000 +0200
@@ -290,7 +290,11 @@
 	       wcsstr wcswcs iswalnum iswspace towlower \
 	       wcslen wcscpy wcsncpy \
 	       uname flock lockf inet_aton inet_addr \
-	       fchmod mkstemp truncate getuid regcomp)
+	       fchmod mkstemp truncate getuid regcomp \
+	       setenv)
+
+dnl we need snprintf/vsnprintf
+NEED_TRIO_CHECK
 
 dnl *****************
 dnl ** common code **
