--- sylpheed-claws-2.2.0/src/etpan/imap-thread.c.socket	2006-05-13 14:56:31.439403000 +0200
+++ sylpheed-claws-2.2.0/src/etpan/imap-thread.c	2006-05-13 14:56:44.709403000 +0200
@@ -8,9 +8,7 @@
 #include <imap.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#if (defined(__DragonFly__) || defined (__NetBSD__) || defined (__FreeBSD__) || defined (__CYGWIN__))
 #include <sys/socket.h>
-#endif
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/wait.h>
