diff -urN less-394.orig/configure.ac less-394.hacked/configure.ac
--- less-394.orig/configure.ac	2005-10-21 18:21:55.000000000 +0200
+++ less-394.hacked/configure.ac	2007-02-24 19:41:02.620000000 +0100
@@ -284,10 +284,14 @@
   [AC_MSG_RESULT(no)])
 fi
 
-AC_MSG_CHECKING(for locale)
-AC_TRY_LINK([#include <locale.h>
-#include <ctype.h>], [setlocale(LC_CTYPE,""); isprint(0); iscntrl(0);],
-  [AC_MSG_RESULT(yes); AC_DEFINE(HAVE_LOCALE)], [AC_MSG_RESULT(no)])
+dnl Verify that we have nl_langinfo and CODESET before checking for locale
+AM_LANGINFO_CODESET
+if test $am_cv_langinfo_codeset = yes; then
+    AC_MSG_CHECKING(for locale)
+    AC_TRY_LINK([#include <locale.h>
+    #include <ctype.h>], [setlocale(LC_CTYPE,""); isprint(0); iscntrl(0);],
+      [AC_MSG_RESULT(yes); AC_DEFINE(HAVE_LOCALE)], [AC_MSG_RESULT(no)])
+fi
 AC_MSG_CHECKING(for ctype functions)
 AC_TRY_LINK([
 #if HAVE_CTYPE_H
