--- cups-1.1.23/systemv/Makefile.pkg	Fri Mar 25 14:51:50 2005
+++ cups-1.1.23/systemv/Makefile	Fri Mar 25 14:55:05 2005
@@ -77,7 +77,7 @@
 	$(INSTALL_BIN) lp $(BINDIR)
 	$(INSTALL_BIN) lpoptions $(BINDIR)
 	$(INSTALL_BIN) lpstat $(BINDIR)
-	-$(INSTALL_BIN) -m 4755 -o $(CUPS_USER) -g $(CUPS_GROUP) lppasswd $(BINDIR)
+	-$(INSTALL_BIN) -m 755 lppasswd $(BINDIR)
 	if test ! -x $(BINDIR)/lppasswd; then \
 		chmod 755 $(BINDIR)/lppasswd; \
 	fi
