--- automake-1.5/lib/am/subdirs.am.subdirs	2001-04-09 07:44:04.000000000 -0700
+++ automake-1.5/lib/am/subdirs.am	2003-04-24 19:27:54.000000000 -0700
@@ -32,7 +32,7 @@ check-recursive installcheck-recursive
 # (2) otherwise, pass the desired values on the `make' command line.
 
 $(RECURSIVE_TARGETS):
-	@set fnord $(MAKEFLAGS); amf=$$2; \
+	@set fnord $$MAKEFLAGS; amf=$$2; \
 	dot_seen=no; \
 	target=`echo $@ | sed s/-recursive//`; \
 	list='$(SUBDIRS)'; for subdir in $$list; do \
@@ -69,7 +69,7 @@ maintainer-clean-recursive
 ## bombs.
 mostlyclean-recursive clean-recursive distclean-recursive \
 maintainer-clean-recursive:
-	@set fnord $(MAKEFLAGS); amf=$$2; \
+	@set fnord $$MAKEFLAGS; amf=$$2; \
 	dot_seen=no; \
 ## For distclean and maintainer-clean we make sure to use the full
 ## list of subdirectories.  We do this so that `configure; make
