--- glib-1.2.10/gmodule/Makefile.am.am16	2007-04-13 11:50:19.056400000 +0200
+++ glib-1.2.10/gmodule/Makefile.am	2007-04-13 11:50:27.076400000 +0200
@@ -42,7 +42,7 @@
 libgplugin_b_la_LIBADD = @G_MODULE_LIBS@ # $(libglib)
 
 noinst_PROGRAMS = testgmodule
-testgmodule_LDFLAGS += @G_MODULE_LDFLAGS@
+testgmodule_LDFLAGS = @G_MODULE_LDFLAGS@
 testgmodule_LDADD = libgmodule.la $(libglib) @G_MODULE_LIBS@
 
 .PHONY: files release
