--- x11vnc-0.9.3/libvncclient/sockets.c.usleep	2008-02-21 08:29:13.320000000 +0100
+++ x11vnc-0.9.3/libvncclient/sockets.c	2008-02-21 08:29:36.280000000 +0100
@@ -43,6 +43,17 @@
 #include <netdb.h>
 #endif
 
+/* XXX usleep(3) is not thread safe on some older systems... */
+extern struct timeval _mysleep;
+#define usleep2(x) \
+            _mysleep.tv_sec  = (x) / 1000000; \
+        _mysleep.tv_usec = (x) % 1000000; \
+        select(0, NULL, NULL, NULL, &_mysleep); 
+#if !defined(X11VNC_USLEEP)
+#undef usleep
+#define usleep usleep2
+#endif
+
 void PrintInHex(char *buf, int len);
 
 rfbBool errorMessageOnReadFailure = TRUE;
