--- samba-3.0.28/source/lib/replace/replace.h.missing	2008-01-16 11:03:42.600000000 +0100
+++ samba-3.0.28/source/lib/replace/replace.h	2008-01-16 11:06:19.360000000 +0100
@@ -336,6 +336,16 @@
 char *rep_mkdtemp(char *template);
 #endif
 
+#ifndef HAVE_PREAD
+#define pread rep_pread
+ssize_t rep_pread(int __fd, void *__buf, size_t __nbytes, off_t __offset);
+#endif
+
+#ifndef HAVE_PWRITE
+#define pwrite rep_pwrite
+ssize_t rep_pwrite(int __fd, const void *__buf, size_t __nbytes, off_t __offset);
+#endif
+
 #ifdef HAVE_LIMITS_H
 #include <limits.h>
 #endif
