--- openssl-0.9.7d/Makefile.org.orig	Sun Jul 11 16:40:20 2004
+++ openssl-0.9.7d/Makefile.org	Sun Jul 11 16:42:12 2004
@@ -460,6 +460,7 @@
 		  (${CC} -v 2>&1 | grep gcc) > /dev/null && WHOLELIB="-Wl,-all,lib$$i.a,-notall"; \
 		  set -x; ${CC} ${SHARED_LDFLAGS} \
 			-shared -o lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
+			-Wl,-rpath,@LIBDIR@ \
 			-Wl,-soname,lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
 			$${WHOLELIB} $$libs ${EX_LIBS} -lc) || exit 1; \
 		libs="-l$$i $$libs"; \
