--- openssl-0.9.7e/fips/rand/fips_rand.c.orig	Sat Nov 27 17:10:20 2004
+++ openssl-0.9.7e/fips/rand/fips_rand.c	Sat Nov 27 17:08:13 2004
@@ -56,7 +56,7 @@
 /* If we don't define _XOPEN_SOURCE_EXTENDED, struct timeval won't
    be defined and gettimeofday() won't be declared with strict compilers
    like DEC C in ANSI C mode.  */
-#define _XOPEN_SOURCE_EXTENDED
+#define _XOPEN_SOURCE_EXTENDED 1
 
 #include <openssl/des.h>
 #include <openssl/rand.h>
