--- openssl-0.9.7c/Makefile.org.orig	2003-12-27 20:21:40.400000000 +0100
+++ openssl-0.9.7c/Makefile.org	2003-12-27 20:22:25.620000000 +0100
@@ -458,8 +458,9 @@
 		fi; \
 		( set -x; ${CC} ${SHARED_LDFLAGS} \
 			-shared -o lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
+			-Wl,-rpath,@LIBDIR@ \
 			-Wl,-soname,lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
-			-all lib$$i.a $$libs ${EX_LIBS} -lc) || exit 1; \
+			-Wl,-all lib$$i.a $$libs ${EX_LIBS} -lc) || exit 1; \
 		libs="-l$$i $$libs"; \
 		done; \
 	fi
