--- wget-1.11.4/src/test.h.orig	2008-04-27 06:48:23.000000000 +0200
+++ wget-1.11.4/src/test.h	2008-09-30 20:06:32.038000000 +0200
@@ -34,8 +34,8 @@
 #define mu_assert(message, test) do { if (!(test)) return message; } while (0)
 #define mu_run_test(test) \
 do { \
-  puts("RUNNING TEST " #test "..."); \
   const char *message = test(); \
+  puts("RUNNING TEST " #test "..."); \
   tests_run++; \
   if (message) return message; \
   puts("PASSED\n"); \
