--- wget-1.10.2/src/sysdep.h.trio	2005-11-12 16:29:52.335219547 +0100
+++ wget-1.10.2/src/sysdep.h	2005-11-12 16:31:15.802324875 +0100
@@ -149,12 +149,6 @@
 #ifndef HAVE_STRPTIME
 char *strptime ();
 #endif
-#ifndef HAVE_SNPRINTF
-int snprintf ();
-#endif
-#ifndef HAVE_VSNPRINTF
-int vsnprintf ();
-#endif
 #ifndef HAVE_MEMMOVE
 void *memmove ();
 #endif
--- wget-1.10.2/src/Makefile.in.trio	2005-11-12 16:31:45.647714130 +0100
+++ wget-1.10.2/src/Makefile.in	2005-11-12 16:32:02.405125322 +0100
@@ -77,7 +77,7 @@
       ftp$o ftp-basic$o ftp-ls$o $(OPIE_OBJ) $(GETOPT_OBJ) hash$o \
       host$o html-parse$o html-url$o http$o $(NTLM_OBJ) init$o    \
       log$o main$o $(MD5_OBJ) netrc$o progress$o ptimer$o recur$o \
-      res$o retr$o safe-ctype$o snprintf$o $(SSL_OBJ) url$o       \
+      res$o retr$o safe-ctype$o $(SSL_OBJ) url$o       \
       utils$o version$o xmalloc$o 
 
 .SUFFIXES:
--- wget-1.10.2/configure.in.trio	2005-11-12 16:32:19.522480894 +0100
+++ wget-1.10.2/configure.in	2005-11-12 16:32:44.058690364 +0100
@@ -218,7 +218,8 @@
 AC_FUNC_MMAP
 AC_FUNC_FSEEKO
 AC_CHECK_FUNCS(strdup strstr strcasecmp strncasecmp strpbrk memmove)
-AC_CHECK_FUNCS(gettimeofday mktime strptime timegm strerror snprintf vsnprintf)
+AC_CHECK_FUNCS(gettimeofday mktime strptime timegm strerror)
+NEED_TRIO_CHECK
 AC_CHECK_FUNCS(usleep select ftello sigblock sigsetjmp signal)
 AC_CHECK_FUNCS(symlink access isatty strtoll strtoimax)
 
