--- wget-1.12/src/sysdep.h.snprintf	2009-09-04 18:31:54.000000000 +0200
+++ wget-1.12/src/sysdep.h	2010-09-11 10:48:41.730000000 +0200
@@ -213,15 +213,6 @@
 void *memrchr (const void *, int, size_t);
 #endif
 
-/* These are defined in snprintf.c.  It would be nice to have an
-   snprintf.h, though.  */
-#ifndef HAVE_SNPRINTF
-int snprintf ();
-#endif
-#ifndef HAVE_VSNPRINTF
-int vsnprintf ();
-#endif
-
 /* Some systems (Linux libc5, "NCR MP-RAS 3.0", and others) don't
    provide MAP_FAILED, a symbolic constant for the value returned by
    mmap() when it doesn't work.  Usually, this constant should be -1.
