--- tiff-v3.5.4/config.site	Sun Sep 26 18:03:46 1999
+++ tiff-v3.5.4/config.site	Tue Jan 18 10:33:09 2000
@@ -41,8 +41,8 @@
 # Package controls.
 #
 #DSO="auto"			# auto|IRIX|IRIX52 enable DSO support for system
-#JPEG="no"			# yes|no configure JPEG support (see below)
-#ZIP="no"			# yes|no configure Deflate support (see below)
+JPEG="yes"			# yes|no configure JPEG support (see below)
+ZIP="yes"			# yes|no configure Deflate support (see below)
 #LIBGL="auto"			# yes|no|auto configure IRIS GL-based tools
 #LIBIMAGE="auto"		# yes|no|auto configure SGI RGB image tools
 #HTML="no"			# yes|no install HTML documentation
@@ -96,6 +96,7 @@
 #
 #DIRS_LIBINC="$DIRS_LIBINC ../src/jpeg-6b" # directory for jpeg include files
 #DIR_JPEGLIB=../src/jpeg-6b		# directory for libjpeg
+DIR_JPEGLIB="/usr/local/lib"
 
 #
 # Deflate-specific parameters; used when Deflate support is enabled (see above).
@@ -111,6 +112,7 @@
 #
 #DIRS_LIBINC="$DIRS_LIBINC ../zlib"	# directory for zlib include files
 #DIR_GZLIB="../zlib"			# directory for libz
+DIR_GZLIB="/usr/local/lib"
 
 #
 # Miscellaneous parameters.
--- tiff-v3.5.4/libtiff/Makefile.in	Sat Jan 29 00:46:31 2000
+++ tiff-v3.5.4/libtiff/Makefile.in	Sat Jan 29 00:46:11 2000
@@ -152,7 +152,7 @@
 	${RANLIB} libtiff.a
 
 $(SHLIB): $(SHOBJS)
-	gcc -o $(SHLIB) -Wl,-rpath,/usr/local/lib -Wl,-soname,$(SONAME) -shared $(SHOBJS)
+	$(CC) -o $(SHLIB) -Wl,-rpath,/usr/local/lib -Wl,-soname,$(SONAME) -shared $(SHOBJS) -ljpeg -lz
 
 #
 # NB: The configure script verifies that the configured
