--- tiff-v3.5.7/libtiff/Makefile.in.orig	2003-12-21 17:00:27.200000000 +0100
+++ tiff-v3.5.7/libtiff/Makefile.in	2003-12-21 17:02:03.890000000 +0100
@@ -171,8 +171,7 @@
 # default IRIX DSO building rule
 IRIXdso: ${OBJS}
 	@if [ "`basename ${CC}`" = "gcc" ]; then \
-	   ${LD} -n32 @ENVOPTS@ -o libtiff.@DSOSUF@ -shared -rdata_shared \
-	   ${OBJS}; \
+	    $(CC) -o $(SHLIB) -Wl,-rpath,/usr/local/lib -Wl,-soname,$(SONAME) -shared $(SHOBJS) -L/usr/local/lib -ljpeg -lz; \
 	else \
 	    ${CC} @ENVOPTS@ -o libtiff.@DSOSUF@ -shared -rdata_shared \
 	        -check_registry ${SRCDIR}/../port/irix/so_locations \
