--- tiff-3.9.6/tools/tiffgt.c.trio	2010-06-08 20:50:44.000000000 +0200
+++ tiff-3.9.6/tools/tiffgt.c	2012-06-10 18:54:25.975975000 +0200
@@ -26,6 +26,11 @@
  */
 
 #include "tif_config.h"
+
+#include <trio.h>
+#define snprintf trio_snprintf
+#define vsnprintf trio_vsnprintf
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- tiff-3.9.6/tools/tiff2pdf.c.trio	2010-12-14 02:45:51.000000000 +0100
+++ tiff-3.9.6/tools/tiff2pdf.c	2012-06-10 18:54:25.965975000 +0200
@@ -26,6 +26,10 @@
 
 #include "tif_config.h"
 
+#include <trio.h>
+#define snprintf trio_snprintf
+#define vsnprintf trio_vsnprintf
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- tiff-3.9.6/tools/Makefile.in.trio	2012-02-18 23:27:08.000000000 +0100
+++ tiff-3.9.6/tools/Makefile.in	2012-06-10 18:55:01.275975000 +0200
@@ -394,7 +394,7 @@
 tiff2bw_SOURCES = tiff2bw.c
 tiff2bw_LDADD = $(LIBTIFF) $(LIBPORT)
 tiff2pdf_SOURCES = tiff2pdf.c
-tiff2pdf_LDADD = $(LIBTIFF) $(LIBPORT)
+tiff2pdf_LDADD = $(LIBTIFF) $(LIBPORT) -ltrio
 tiff2ps_SOURCES = tiff2ps.c
 tiff2ps_LDADD = $(LIBTIFF) $(LIBPORT)
 tiff2rgba_SOURCES = tiff2rgba.c
@@ -419,7 +419,7 @@
 tiffsplit_LDADD = $(LIBTIFF) $(LIBPORT)
 tiffgt_SOURCES = tiffgt.c
 tiffgt_CFLAGS = $(CFLAGS) $(GLUT_CFLAGS) $(AM_CFLAGS)
-tiffgt_LDADD = $(LIBTIFF) $(LIBPORT) $(X_LIBS) $(GLUT_LIBS)
+tiffgt_LDADD = $(LIBTIFF) $(LIBPORT) $(X_LIBS) $(GLUT_LIBS) -ltrio
 INCLUDES = -I../libtiff -I$(top_srcdir)/libtiff
 all: all-am
 
