--- tiff-v3.5.4/libtiff/Makefile.in.shlib	Sun Nov 28 15:15:36 1999
+++ tiff-v3.5.4/libtiff/Makefile.in	Wed Dec 29 16:28:14 1999
@@ -64,6 +64,8 @@
 COPTS	= @GCOPTS@
 OPTIMIZER=-O
 CFLAGS	= @ENVOPTS@ @LIBCOPTS@ ${COPTS} ${OPTIMIZER} ${IPATH} ${CONF_LIBRARY}
+SONAME = libtiff.so.3
+SHLIB = libtiff.so.3.5
 #
 SRCS	= \
 	tif_aux.c \
@@ -135,7 +137,8 @@
 	tif_write.o \
 	tif_zip.o \
 	${NULL}
-TARGETS	= libtiff.a
+TARGETS	= libtiff.a $(SHLIB)
+SHOBJS= $(OBJS:%.o=%.so) 
 
 all:	${TARGETS}
 	if [ @DSO@dso != nodso ]; then \
@@ -148,6 +151,9 @@
 	${AR} ${AROPTS} libtiff.a $?
 	${RANLIB} libtiff.a
 
+$(SHLIB): $(SHOBJS)
+	gcc -o $(SHLIB) -Wl,-rpath,/usr/local/lib -Wl,-soname,$(SONAME) -shared $(SHOBJS)
+
 #
 # NB: The configure script verifies that the configured
 # tools are capable of producing a DSO before enabling
@@ -346,3 +352,6 @@
 	    mkg3states tif_fax3sm.c \
 	    mkversion version.h \
 	    libtiff.a libtiff.@DSOSUF@ libtiff.@DSOSUF_VERSION@ *dso
+
+%.so : %.c
+	$(CC) -c -fpic $(CFLAGS) $(CPPFLAGS) $< -o $@
--- tiff-v3.5.4/tools/Makefile.in.shlib	Mon Dec 20 18:46:11 1999
+++ tiff-v3.5.4/tools/Makefile.in	Wed Dec 29 16:30:44 1999
@@ -47,10 +47,10 @@
 IPATH	= -I. -I${SRCDIR} -I${LIBDIR}
 CFLAGS	= @ENVOPTS@ ${COPTS} ${OPTIMIZER} ${IPATH}
 #
-TIFFLIB	= ${DEPTH}/libtiff/libtiff.@DSOSUF@
+TIFFLIB	= ${DEPTH}/libtiff/libtiff.so.3.5
 LIBJPEG	= @LIBJPEG@
 LIBGZ	= @LIBGZ@
-LIBTIFF	= @TIFFLIBREF@
+LIBTIFF	= -L${DEPTH}/libtiff  -ltiff
 LIBPORT	= @LIBPORT@
 MACHLIBS= @MACHDEPLIBS@
 LIBS	= ${LIBTIFF} ${LIBJPEG} ${LIBGZ} ${LIBPORT} ${MACHLIBS}
